### PR TITLE
vscode page and other docs moves

### PR DESCRIPTION
**Pull request by thisisjofrank**

### CODE REVIEW

File: runtime/reference/vscode.md [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+        "mod.ts"
+      ],
+      "problemMatcher": [
+        "$deno"
+      ],
+      "label": "deno: run"
+    }
+  ]
+}
+```
```
Comment (marvinhagemeister) [inline runtime/reference/vscode.md:307]: I don't understand this whole section. Is it talking about `deno task` or how to set up tasks in vscode in general for their debugging feature?

File: runtime/reference/vscode.md [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+[multi-root workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces)
+configuration, where certain settings can be applied to workspace folders within
+a workspace.
+
+When you add folders to your workspace and open the settings, you will have
+access to the per folder settings. If you look at the `.vscode/settings.json` in
+a folder, you will see a visual indication of what settings apply to folder,
+versus those that come from the workspace configuration:
+
+![screenshot of the .vscode/setting.json configured as a workspace folder](./images/workspace_folder_config.png)
```
Comment (marvinhagemeister) [inline runtime/reference/vscode.md:322]: yay 👍

File: runtime/reference/vscode.md [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+
+Once added, VS Code will prompt if you want to open the project in a dev
+container. If you choose to, VS Code will build the development container and
+re-open the workspace using the development container, which will have the Deno
+CLI and the `vscode_deno` extension installed in it.
+
+## Troubleshooting
+
+The following sections cover challenges you might face when using the extension
+and try to give likely causes.
```
Comment (marvinhagemeister) [inline runtime/reference/vscode.md:393]: This is a great vscode feature, but not sure if we need to describe it in our docs if it's a "just works" with Deno thing.

File: runtime/reference/vscode.md [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+Test will be displayed in the Test Explorer at the top level with the module
+that contains the test. Inside the module will be all the tests that have been
+discovered, and if you are using test steps, they will be included under the
+test.
+
+In most cases, the Deno language server will be able to statically identify
+tests, but if you are generating tests dynamically, Deno may not be aware of
+them until runtime. In these cases it may not be possible to filter these tests
+out of a run, but they will be added to the explorer view as they are
+encountered.
```
Comment (marvinhagemeister) [inline runtime/reference/vscode.md:486]: We had a section about testing in vscode earlier in the document. I feel like most of this here is duplicative and can be folded into the earlier section. What's new here is a mention of the test explorer. We can drop the "console.log's show up in the terminal output" part imo.

**Review by marvinhagemeister — COMMENTED** [review]
Great pass on unifying these. I like the overall structure. Left some comments where some information is not up to date.

File: runtime/_data.ts [hunk 5]
```diff
@@ -160,25 +160,17 @@ export const sidebar = [
...
+        id: "/runtime/reference/vscode/",
       },
+      "/runtime/reference/private_repositories/",
+      "/runtime/reference/tcp_udp_connections/",
     ],
   },
   {
     title: "Advanced Topics",
     items: [
-      "/runtime/manual/advanced/private_repositories/",
```
[the remark this answers is not in the document]
Comment (thisisjofrank) [reply runtime/_data.ts:181]: it should now be in the references folder

File: runtime/reference/vscode.md [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+where you can choose to install.
+
+Once you install the extension for the first time, you should receive a splash
+page that welcomes you to the extension. (If you missed it, or want to see it
+again, <kbd>⌘</kbd> <kbd>⇧</kbd> <kbd>P</kbd> to open the command palette, and
+run the `Deno: Welcome` command.)
+
+## Configuring the extension
+
+### Deno enabling a workspace
```
[the remark this answers is not in the document]
Comment (thisisjofrank) [reply runtime/reference/vscode.md:33]: Should I update it to **Enabling Deno in a VS Code workspace** maybe?

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**